lib/system/database.rb

```diff
@@ -148,22 +148,25 @@ def adapter_module
 end
 
 ActiveSupport.on_load(:active_record) do
-if System::Database.oracle? && defined?(ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks)
-  ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks.class_eval do
-    prepend(Module.new do
-
-      def create
-        super
-        connection.execute "GRANT create trigger TO #{username}"
-        connection.execute "GRANT create procedure TO #{username}"
-      end
-
-      protected
-
-      def username
-        @config['username']
-      end
-    end)
+  if System::Database.oracle? && defined?(ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks)
+    ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks.class_eval do
+      prepend(Module.new do
+        # If ORACLE_SYSTEM_PASSWORD is provided, Porta impersonates Oracle's SYSTEM user to create/update a non-SYSTEM
+        # user and grants it permissions. This is a behaviour that should preferably be used during development as
+        # the usage of Oracle's SYSTEM user grants the application more power than it should have in the first place.
+        # Alternatively, Porta should be provided with a non-SYSTEM user that has been granted the necessary
+        # permissions.
+        def create
+          if ENV['ORACLE_SYSTEM_PASSWORD'].present?
+            super
+            connection.execute "GRANT create trigger TO #{username}"
+            connection.execute "GRANT create procedure TO #{username}"
+          else
+            Rails.logger.warn('Since ORACLE_SYSTEM_PASSWORD was not provided, no new database will be created')
```

Comment:
Hm, I am not sure I like this warning. It looks like `ORACLE_SYSTEM_PASSWORD` is something that is **supposed** to be provided, and as it's not there is some undesired outcome (DB will not be created).

I would probably prefer to do it the other way around - warn when `ORACLE_SYSTEM_PASSWORD` IS provided and inform that the Oracle SYSTEM user will be used to update the non-system user and create the database (or just set up? what happens if the DB already exists?).

Reply:
>(or just set up? what happens if the DB already exists?).

It still tries to create/alter the non-SYSTEM user and grant it permissions. Nothing changes on the database if it already exists.

Reply:
Changed in 353f9f129.